### PR TITLE
Update dependencies after branching for v2.7

### PR DESCRIPTION
**Pull request by prachidamle**
Updating the dependencies in rancher/rancher after branches for v2.7 are created.

Versions for UI, DASHBOARD and CLI will be the same as 2.6 until new releases are made to these

Updated RKE to latest release  v1.3.14

### CODE REVIEW

**Review by aiyengar2 — APPROVED** [review]

File: package/Dockerfile [hunk 1]
```diff
@@ -15,12 +15,14 @@ WORKDIR /var/lib/rancher
 
 ARG ARCH=amd64
 ARG IMAGE_REPO=rancher
+# System-charts will continue to point to dev-v2.6 branch since eventually in 2.7.x the plan is
+# to deprecate the v1 tools and remove this dependency
```
Comment (deniseschannon) [inline package/Dockerfile:19]: I'd update to use the language "remove the v1 tools"

File: scripts/package [hunk 2]
```diff
@@ -5,9 +5,11 @@ source $(dirname $0)/version
 
 ARCH=${ARCH:-"amd64"}
 SYSTEM_CHART_REPO_DIR=build/system-charts
+# System-charts will continue to point to dev-v2.6 branch since eventually in 2.7.x the plan is
+# to deprecate v1 tools and remove this dependency
```
Comment (deniseschannon) [inline scripts/package:9]: update to "to remove v1 tools"

**Review by deniseschannon — CHANGES_REQUESTED** [review]
What about this reference? Should this be set at all based on the comment? 

https://github.com/rancher/rancher/blob/release/v2.7/Dockerfile.dapper#L15

File: package/Dockerfile (hunk 1 above)
Comment (prachidamle) [reply package/Dockerfile]: updated

File: scripts/package (hunk 2 above)
Comment (prachidamle) [reply scripts/package]: updated

**Comment by prachidamle** [comment]
> What about this reference? Should this be set at all based on the comment?
> 
> https://github.com/rancher/rancher/blob/release/v2.7/Dockerfile.dapper#L15

@deniseschannon updated that reference to dev-v2.7 as well. 
Was not sure if we need it anymore from the comment in the checklist about 2.4 and kdm vendoring; but will keep it updated to not break anything

**Review by aiyengar2 — APPROVED** [review]

**Review by deniseschannon — APPROVED** [review]

**Comment by prachidamle** [comment]
Validated that the CI drone check has passed, merging in the PR.

**Pull request merged**